File: java/LocationOutilsProcedural.java
import java.util.Scanner;

public class LocationOutilsProcedural {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);


        String[] nomsOutils = {
                "Perceuse",
                "Ponceuse",
                "Tondeuse",
                "Marteau piqueur",
                "Echelle"
        };
        double[] prixParJour = {
                15.0,
                12.5,
                25.0,
                40.0,
                10.0
        };
        boolean[] disponible = {
                true,
                true,
                true,
                true,
                true
        };

        boolean quitter = false;

        while (!quitter) {
            System.out.println("=====================================");
            System.out.println("   Mini systeme de location d'outils ");
            System.out.println("=====================================");
            System.out.println("1. Lister les outils");
            System.out.println("2. Louer un outil");
            System.out.println("3. Retourner un outil");
            System.out.println("4. Quitter");
            System.out.print("Votre choix : ");

            String ligne = scanner.nextLine();
            int choix;
            try {
                choix = Integer.parseInt(ligne);
            } catch (NumberFormatException e) {
                System.out.println("Choix invalide.");
                continue;
            }

            if (choix == 1) {
                System.out.println("\nListe des outils :");
                for (int i = 0; i < nomsOutils.length; i++) {
                    System.out.println(i + " - " + nomsOutils[i]
                            + " | " + prixParJour[i] + " €/jour"
                            + " | " + (disponible[i] ? "DISPONIBLE" : "LOUE"));
                }
                System.out.println();

            } else if (choix == 2) {
                System.out.print("\nEntrez l'index de l'outil a louer : ");
                String indexStr = scanner.nextLine();
                int index;
                try {
                    index = Integer.parseInt(indexStr);
                } catch (NumberFormatException e) {
                    System.out.println("Index invalide.\n");
                    continue;
                }

                if (index < 0 || index >= nomsOutils.length) {
                    System.out.println("Aucun outil pour cet index.\n");
                    continue;
                }

                if (!disponible[index]) {
                    System.out.println("Outil deja loue. Impossible de le louer a nouveau.\n");
                    continue;
                }

                System.out.print("Nombre de jours de location : ");
                String joursStr = scanner.nextLine();
                int nbJours;
                try {
                    nbJours = Integer.parseInt(joursStr);
                } catch (NumberFormatException e) {
                    System.out.println("Nombre de jours invalide.\n");
                    continue;
                }

                if (nbJours <= 0) {
                    System.out.println("Le nombre de jours doit etre strictement positif.\n");
                    continue;
                }

                double prixTotal = nbJours * prixParJour[index];
                disponible[index] = false;

                System.out.println("\n---------- Recapitulatif ----------");
                System.out.println("Outil : " + nomsOutils[index]);
                System.out.println("Duree : " + nbJours + " jour(s)");
                System.out.println("Prix par jour : " + prixParJour[index] + " €");
                System.out.println("Prix total : " + prixTotal + " €");
                System.out.println("-----------------------------------\n");

            } else if (choix == 3) {
                System.out.print("\nEntrez l'index de l'outil a retourner : ");
                String indexStr = scanner.nextLine();
                int index;
                try {
                    index = Integer.parseInt(indexStr);
                } catch (NumberFormatException e) {
                    System.out.println("Index invalide.\n");
                    continue;
                }

                if (index < 0 || index >= nomsOutils.length) {
                    System.out.println("Aucun outil pour cet index.\n");
                    continue;
                }

                if (disponible[index]) {
                    System.out.println("Cet outil est deja marque comme disponible.\n");
                    continue;
                }

                disponible[index] = true;
                System.out.println("Outil " + nomsOutils[index] + " retourne. Il est maintenant disponible.\n");

            } else if (choix == 4) {
                quitter = true;
                System.out.println("Au revoir !");
            } else {
                System.out.println("Choix invalide.\n");
            }
        }

        scanner.close();
    }
}

File: php/location_outils_procedural.php
<?php

$toolNames = [
    "Perceuse",
    "Ponceuse",
    "Tondeuse",
    "Marteau piqueur",
    "Echelle"
];

$pricesPerDay = [
    15.0,
    12.5,
    25.0,
    40.0,
    10.0
];

$available = [
    true,
    true,
    true,
    true,
    true
];

function readLineInput(string $prompt): string {
    echo $prompt;
    $line = fgets(STDIN);
    if ($line === false) {
        return "";
    }
    return trim($line);
}

$quit = false;

while (!$quit) {
    echo "=====================================\n";
    echo "   Mini systeme de location d'outils \n";
    echo "=====================================\n";
    echo "1. Lister les outils\n";
    echo "2. Louer un outil\n";
    echo "3. Retourner un outil\n";
    echo "4. Quitter\n";

    $choiceLine = readLineInput("Votre choix : ");
    if (!ctype_digit($choiceLine)) {
        echo "Choix invalide.\n\n";
        continue;
    }
    $choice = (int)$choiceLine;

    if ($choice === 1) {
        echo "\nListe des outils :\n";
        for ($i = 0; $i < count($toolNames); $i++) {
            $status = $available[$i] ? "DISPONIBLE" : "LOUE";
            echo $i . " - " . $toolNames[$i]
                . " | " . $pricesPerDay[$i] . " €/jour"
                . " | " . $status . "\n";
        }
        echo "\n";

    } elseif ($choice === 2) {
        $indexLine = readLineInput("\nEntrez l'index de l'outil a louer : ");
        if (!ctype_digit($indexLine)) {
            echo "Index invalide.\n\n";
            continue;
        }
        $index = (int)$indexLine;

        if ($index < 0 || $index >= count($toolNames)) {
            echo "Aucun outil pour cet index.\n\n";
            continue;
        }

        if ($available[$index] === false) {
            echo "Outil deja loue. Impossible de le louer a nouveau.\n\n";
            continue;
        }

        $daysLine = readLineInput("Nombre de jours de location : ");
        if (!ctype_digit($daysLine)) {
            echo "Nombre de jours invalide.\n\n";
            continue;
        }
        $days = (int)$daysLine;

        if ($days <= 0) {
            echo "Le nombre de jours doit etre strictement positif.\n\n";
            continue;
        }

        $totalPrice = $days * $pricesPerDay[$index];
        $available[$index] = false;

        echo "\n---------- Recapitulatif ----------\n";
        echo "Outil : " . $toolNames[$index] . "\n";
        echo "Duree : " . $days . " jour(s)\n";
        echo "Prix par jour : " . $pricesPerDay[$index] . " €\n";
        echo "Prix total : " . $totalPrice . " €\n";
        echo "-----------------------------------\n\n";

    } elseif ($choice === 3) {
        // Retourner un outil
        $indexLine = readLineInput("\nEntrez l'index de l'outil a retourner : ");
        if (!ctype_digit($indexLine)) {
            echo "Index invalide.\n\n";
            continue;
        }
        $index = (int)$indexLine;

        if ($index < 0 || $index >= count($toolNames)) {
            echo "Aucun outil pour cet index.\n\n";
            continue;
        }

        if ($available[$index] === true) {
            echo "Cet outil est deja marque comme disponible.\n\n";
            continue;
        }

        $available[$index] = true;
        echo "Outil " . $toolNames[$index] . " retourne. Il est maintenant disponible.\n\n";

    } elseif ($choice === 4) {
        $quit = true;
        echo "Au revoir !\n";
    } else {
        echo "Choix invalide.\n\n";
    }
}
